site/js/Actor.js

```diff
@@ -0,0 +1,41 @@
+// See https://en.wikipedia.org/wiki/Actor_model, ish.
+// Like an EventEmitter but cooler.
+
+export default class Actor {
+  constructor() {
+    this.__subscribers = []
+  }
+
+  init() {} // Subscribe to events here.
+  go() {}   // Perform initial actions here.
+
+  // Subscribes to a message type from this actor.
+  on(message, callback, times = Infinity) {
+    let subs = this.__subscribers[message] || []
+    subs.push({ callback, times })
+
+    this.__subscribers[message] = subs
+  }
+
+  // Waits for `message` and then resolves the promise.
+  waitFor(message) {
+    return new Promise(resolve => {
+      this.on(message, resolve, 1)
+    })
+  }
+
+  // Emits a message and notifies all subscribers.
+  // Other actors should not call this.
+  emit(message, ...data) {
+    let subs = (this.__subscribers[message] || [])
+
+    console.info(this.name + '::', message, ...data)
```

Comment:
In the future it might be a good idea to have a log-level system.. for now this is definitely handy for debugging, but it could become console-spammy in the future.